### PR TITLE
Restrict user permission

### PR DESCRIPTION
**Pull request by rhnonose**
- Setup user permission with canada and canary
- Restrict volunteers' permission

### CODE REVIEW

File: test/controllers/task_controller_test.exs [hunk 1]
```diff
@@ -89,6 +90,8 @@ defmodule CoursePlanner.TaskControllerTest do
     {:ok, volunteer} = Volunteers.new(@volunteer, "whatever")
     task = Map.put(@valid_attrs, :user_id, volunteer.id)
     {:ok, task} = Tasks.new(task)
+    # IO.inspect volunteer
```
Comment (ggpasqualino) [inline test/controllers/task_controller_test.exs]: could you remove the commented `IO.inspect`?

File: lib/course_planner/permissions.ex [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+  def can?(%User{role: "Coordinator"}, _, _), do: true
+  def can?(%User{role: "Volunteer", id: id}, :done, %Task{user_id: id}), do: true
+  def can?(%User{role: "Volunteer", id: id}, :show, %Task{user_id: id}), do: true
+  def can?(%User{role: "Volunteer"}, :index, Task), do: true
+  def can?(%User{role: "Volunteer"}, :grab, %Task{user_id: nil}), do: true
+  def can?(%User{role: "Volunteer"}, :show, %Task{user_id: nil}), do: true
+  def can?(%User{role: "Volunteer"}, _, Task), do: false
+  def can?(%User{role: "Volunteer"}, _, %Task{}), do: false
+  def can?(_, _, _), do: true
```
Comment (ggpasqualino) [inline lib/course_planner/permissions.ex]: this means that for now the students and teachers can do everything, right?

File: test/lib/course_planner/permissions_test.exs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    role: "Coordinator"
+  }
+  @volunteer %User{
+    id: 2,
+    email: "valid@email2",
+    role: "Volunteer"
+  }
+
+  Enum.map([:index, :show, :create, :update, :delete, :edit], fn action ->
+    @action action
```
Comment (ggpasqualino) [inline test/lib/course_planner/permissions_test.exs]: why do you need this `@action action`?

File: lib/course_planner/permissions.ex (hunk 2 above)
Comment (rhnonose) [inline lib/course_planner/permissions.ex]: Yes because it's the system's current behavior.

File: web/controllers/task_controller.ex [hunk 4]
```diff
@@ -5,6 +5,10 @@ defmodule CoursePlanner.TaskController do
   alias CoursePlanner.Tasks.Task
   alias CoursePlanner.Volunteers
 
+  import Canary.Plugs
+  plug :authorize_resource, model: Task, id_name: "task_id", only: [:grab, :done]
+  plug :authorize_resource, model: Task, except: [:grab, :done]
```
Comment (ggpasqualino) [inline web/controllers/task_controller.ex]: the `:done` action doesn't exist anymore

**Pull request merged**